File: OnlineEditorsExample/OnlineEditorsExampleRuby/app/models/file_utility.rb
class FileUtility

  @@exts_document = %w(.docx .doc .odt .rtf .txt .html .htm .mht .pdf .djvu .fb2 .epub .xps)

  @@exts_spreadsheet = %w(.xls .xlsx .ods .csv)

  @@exts_presentation = %w(.ppt .pptx .odp)

  class << self

    def get_file_type(file_name)
        ext = File.extname(file_name)

        if @@exts_document.include? ext
          return 'text'
        end

        if @@exts_spreadsheet.include? ext
          return 'spreadsheet'
        end

        if @@exts_presentation.include? ext
          return 'presentation'
        end

        'text'
    end

  end

end

File: OnlineEditorsExample/OnlineEditorsExampleMVC/Models/FileUtility.cs
using System.Collections.Generic;
using System.IO;

namespace OnlineEditorsExampleMVC.Models
{
    public static class FileUtility
    {
        public enum FileType
        {
            Text,
            Spreadsheet,
            Presentation
        }

        public static FileType GetFileType(string fileName)
        {
            var ext = Path.GetExtension(fileName).ToLower();

            if (ExtsDocument.Contains(ext)) return FileType.Text;
            if (ExtsSpreadsheet.Contains(ext)) return FileType.Spreadsheet;
            if (ExtsPresentation.Contains(ext)) return FileType.Presentation;

            return FileType.Text;
        }

        public static readonly List<string> ExtsDocument = new List<string>
            {
                ".docx", ".doc", ".odt", ".rtf", ".txt",
                ".html", ".htm", ".mht", ".pdf", ".djvu",
                ".fb2", ".epub", ".xps"
            };

        public static readonly List<string> ExtsSpreadsheet = new List<string>
            {
                ".xls", ".xlsx",
                ".ods", ".csv"
            };

        public static readonly List<string> ExtsPresentation = new List<string>
            {
                ".ppt", ".pptx",
                ".odp"
            };
    }
}